sigopt/factory.py

```diff
@@ -48,6 +51,18 @@ def _on_experiment_created(self, experiment):
       experiment.id,
     )
 
+  def create_project(self, project_name):
+    client_id = get_client_id(self.connection)
+    try:
+      project = self.connection.clients(client_id).projects().create(id=self.project, name=project_name)
```

Comment:
Having trouble following - when/how does `self.project` get set for `create_+project` calls? Or it usually) doesn't and just sends `None` which will get converted on the backend to a safe repr of `project_name`?

Reply:
`self.project` is a convenience property that gives read-only access to `self._project_id`. `self._project_id` is always set in the constructor for `SigOptFactory`.